conans/client/generators/b2.py

```diff
@@ -0,0 +1,420 @@
+                '<define>$(defines({name},{variation}))'.format(name=name, variation=self.b2_variation_id),
+                '<cflags>$(cflags({name},{variation}))'.format(name=name, variation=self.b2_variation_id),
+                '<cxxflags>$(cppflags({name},{variation}))'.format(name=name, variation=self.b2_variation_id),
+                '<link>shared:<linkflags>$(sharedlinkflags({name},{variation}))'.format(name=name, variation=self.b2_variation_id)
+                ])
+
+        if user:
+            for uk, uv in user.vars.items():
+                result += self.b2_constant(uk.lower() + ',' + name, 'user', [uv])
+
+        return result
+
+    def b2_targets_for_dep(self, name, info):
+        '''
+        Generates individual targets for the libraries in a package and a single "libs"
+        collective alias target that refers to them.
+        '''
+        if not info:
+            return []
+        name = name.lower()
+        result = []
+        if info.libs:
+            for lib in info.libs:
+                result += [self.conanbuildinfo_variation_lib_template.format(
+                    name=name, lib=lib, variation=self.b2_variation_id)]
+            result += [self.conanbuildinfo_variation_alias_template.format(
+                name=name, libs=" ".join(info.libs), variation=self.b2_variation_id)]
+        else:
+            result += [self.conanbuildinfo_variation_alias_template.format(
+                name=name, libs="", variation=self.b2_variation_id)]
+
+        return result
+
+    def b2_constant(self, name, var, val, is_paths=False):
+        '''
+        Generates a constant definition for the given variable and value(s). If is_path
+        is True the value(s) are reformated to be acceptable to b2.
+        '''
+        if not val:
+            return []
+        if is_paths:
+            val = list(self.b2_path(p) for p in val)
+        value = []
+        for v in val:
+            if v.startswith('<'):
+                value += ['    {val}'.format(val=v)]
+            else:
+                value += ['    "{val}"'.format(val=v)]
+        return [self.conanbuildinfo_variation_constant_template.format(
+            name=name, var=var, variation=self.b2_variation_id, value="\n".join(value)
+        )]
+
+    def b2_path(self, p):
+        '''
+        Adjust a ragular path to the form b2 can use in source code.
+        '''
+        return p.replace('\\', '/')
+
+    def b2_features(self, m):
+        '''
+        Generated a b2 requirements list, i.e. <name>value list, from the given
+        map of key-values.
+        '''
+        result = []
+        for k, v in sorted(m.items()):
+            if v:
+                result += ['<%s>%s' % (k, v)]
+        return result
+
+    @property
+    def conanbuildinfo_variation_jam(self):
+        return 'conanbuildinfo-%s.jam'%(self.b2_variation_key)
+
+    _b2_variation_key = None
```

Comment:
it is a bit noisy to have this here. Either declare them at the class level (just after the class declaration line), or better, use the idiom ``getattr(self, "_b2_variation_key", None)`` in the method.